src/Blob.mo

```diff
@@ -1,53 +1,199 @@
+  /// the bytes. Returns the `Order` (either `#less`, `#equal`, or `#greater`)
+  /// by comparing `blob1` with `blob2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob1 = "\00\00\00" : Blob;
+  /// let blob2 = "\00\FF\00" : Blob;
+  /// Blob.compare(blob1, blob2) // => #less
+  /// ```
+  public func compare(blob1 : Blob, blob2 : Blob) : { #less; #equal; #greater } {
+    if (blob1 < blob2) {
+      #less
+    } else if (blob1 == blob2) {
+      #equal
+    } else {
+      #greater
+    }
   };
 
-  /// Creates a blob from an array of bytes, by copying each element.
-  public let fromArray : [Nat8] -> Blob = Prim.arrayToBlob;
+  /// Equality function for `Blob` types.
+  /// This is equivalent to `blob1 == blob2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob1 = "\00\FF\00" : Blob;
+  /// let blob2 = "\00\FF\00" : Blob;
+  /// ignore Blob.equal(blob1, blob2);
+  /// blob1 == blob2 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `==` operator) is so that you can use it as a higher order
+  /// function. It is not possible to use `==` as a higher order function at the moment.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// import Buffer "mo:base/Buffer";
+  ///
+  /// let buffer1 = Buffer.Buffer<Blob>(3);
+  /// let buffer2 = Buffer.Buffer<Blob>(3);
+  /// Buffer.equal(buffer1, buffer2, Blob.equal) // => true
+  /// ```
+  public func equal(blob1 : Blob, blob2 : Blob) : Bool { blob1 == blob2 };
+
+  /// Inequality function for `Blob` types.
+  /// This is equivalent to `blob1 != blob2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob1 = "\00\AA\AA" : Blob;
+  /// let blob2 = "\00\FF\00" : Blob;
+  /// ignore Blob.notEqual(blob1, blob2);
+  /// blob1 != blob2 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `!=` operator) is so that you can use it as a higher order
+  /// function. It is not possible to use `!=` as a higher order function at the moment.
+  public func notEqual(blob1 : Blob, blob2 : Blob) : Bool { blob1 != blob2 };
 
-  /// Creates a blob from a mutable array of bytes, by copying each element.
-  public let fromArrayMut : [var Nat8] -> Blob = Prim.arrayMutToBlob;
+  /// 'Less than' function for `Blob` types.
+  /// This is equivalent to `blob1 < blob2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob1 = "\00\AA\AA" : Blob;
+  /// let blob2 = "\00\FF\00" : Blob;
+  /// ignore Blob.less(blob1, blob2);
+  /// blob1 < blob2 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `<` operator) is so that you can use it as a higher order
+  /// function. It is not possible to use `<` as a higher order function at the moment.
```

Comment:
```suggestion
  /// to the existing `<` operator) is so that you can use it as a function value.
  /// It is not possible to use `<` as a function value at the moment.
```